sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OffHeapColumnVector.java

```diff
@@ -165,6 +165,12 @@ public byte getByte(int rowId) {
     }
   }
 
+  @Override
+  public void getBytes(byte[] dst, int rowId, int count) {
+    assert(dictionary == null);
```

Comment:
We could support a case with dictionary. In my opinion, we may perform element-wise copy or add new API to `dictionary` for bulk decoding.